### PR TITLE
WIP: Clean setup

### PR DESCRIPTION
**Pull request by davidungar**
Reorganize CompilerInstance::setup() on the way to multiple primaries.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test OS X platform

**Comment by davidungar** [comment]
@swift-ci please smoke test OS X platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

File: tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp [hunk 1]
```diff
@@ -761,27 +754,27 @@ ASTUnitRef ASTProducer::createASTUnit(SwiftASTManager::Implementation &MgrImpl,
...
+                                                             return;
+                                                           
+                                                           auto Content = MgrImpl.getFileContent(File, Error);
+                                                           if (!Content.Buffer) {
+                                                             LOG_WARN_FUNC("failed getting file contents for " << File << ": " << Error);
+                                                             // File may not exist, continue and recover as if it was empty.
+                                                             Content.Buffer = llvm::MemoryBuffer::getNewMemBuffer(0, File);
+                                                           }
+                                                           Contents.push_back(std::move(Content));
+                                                         });
```
Comment (compnerd) [inline tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp]: Just a drive-by.  This is pretty far indented.  Would be nice to just hoist this out and create a local variable for the lambda which you can pass as an argument.  It would significantly un-indent the code here.  There is at least one other instance of this.

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test os x platform
